api/v1/helper_test.go

```diff
@@ -1122,3 +1122,415 @@ func TestNeedToUpdateSriov(t *testing.T) {
+		expectedBridges v1.Bridges
+		expectedErr     bool
+	}{
+		{
+			tname:        "no selectors",
+			currentState: newNodeState(),
+			policy: &v1.SriovNetworkNodePolicy{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "p1",
+				},
+				Spec: v1.SriovNetworkNodePolicySpec{
+					DeviceType:  consts.DeviceTypeNetDevice,
+					NicSelector: v1.SriovNetworkNicSelector{},
+					NodeSelector: map[string]string{
+						"feature.node.kubernetes.io/network-sriov.capable": "true",
+					},
+					NumVfs:       2,
+					Priority:     99,
+					EswitchMode:  "switchdev",
+					ResourceName: "p1res",
+					Bridge:       v1.Bridge{OVS: &v1.OVSConfig{}},
+				},
+			},
+			expectedBridges: v1.Bridges{},
+		},
+		{
+			tname:        "not switchdev config",
+			currentState: newNodeState(),
+			policy: &v1.SriovNetworkNodePolicy{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "p1",
+				},
+				Spec: v1.SriovNetworkNodePolicySpec{
+					DeviceType: consts.DeviceTypeNetDevice,
+					NicSelector: v1.SriovNetworkNicSelector{
+						RootDevices: []string{"0000:86:00.0"},
+					},
+					NodeSelector: map[string]string{
+						"feature.node.kubernetes.io/network-sriov.capable": "true",
+					},
+					NumVfs:       2,
+					Priority:     99,
+					EswitchMode:  "legacy",
+					ResourceName: "p1res",
+					Bridge:       v1.Bridge{OVS: &v1.OVSConfig{}},
+				},
+			},
+			expectedBridges: v1.Bridges{},
+			expectedErr:     true,
+		},
+		{
+			tname:        "bad linkType",
+			currentState: newNodeState(),
+			policy: &v1.SriovNetworkNodePolicy{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "p1",
+				},
+				Spec: v1.SriovNetworkNodePolicySpec{
+					DeviceType: consts.DeviceTypeNetDevice,
+					NicSelector: v1.SriovNetworkNicSelector{
+						RootDevices: []string{"0000:86:00.0"},
+					},
+					NodeSelector: map[string]string{
+						"feature.node.kubernetes.io/network-sriov.capable": "true",
+					},
+					NumVfs:       2,
+					Priority:     99,
+					EswitchMode:  "switchdev",
+					LinkType:     "ib",
+					ResourceName: "p1res",
+					Bridge:       v1.Bridge{OVS: &v1.OVSConfig{}},
+				},
+			},
+			expectedBridges: v1.Bridges{},
+			expectedErr:     true,
+		},
+		{
+			tname:        "externally managed",
+			currentState: newNodeState(),
+			policy: &v1.SriovNetworkNodePolicy{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "p1",
+				},
+				Spec: v1.SriovNetworkNodePolicySpec{
+					DeviceType: consts.DeviceTypeNetDevice,
+					NicSelector: v1.SriovNetworkNicSelector{
+						RootDevices: []string{"0000:86:00.0"},
+					},
+					NodeSelector: map[string]string{
+						"feature.node.kubernetes.io/network-sriov.capable": "true",
+					},
+					NumVfs:            2,
+					Priority:          99,
+					EswitchMode:       "switchdev",
+					ExternallyManaged: true,
+					ResourceName:      "p1res",
+					Bridge:            v1.Bridge{OVS: &v1.OVSConfig{}},
+				},
+			},
+			expectedBridges: v1.Bridges{},
+			expectedErr:     true,
```

Comment:
is externally manage not support at all or right now and will be in the future?

Reply:
If the device is externally managed we are not managing bridges for it, but It is possible to create OVSNetwork for Externally managed PF to announce "manually" created resources. For externally managed PFs we are not changing PF-level settings. For me "software bridge config for PF" option is a PF-level settings and I think we should not change it externally managed for PFs.